### ISSUE TITLE
Not getting BucketSizeBytes from dynamic S3 metrics (but getting NumberOfObjects OK)

### ISSUE THREAD

**Issue by abuchananTW** [ivx/yet-another-cloudwatch-exporter#46]
Config file:
```
    discovery:
      exportedTagsOnMetrics:
        ec2:
          - Name
        elb:
          - Name
        es:
          - Name
      jobs:
      - region: us-east-2
        type: "ec2"
        searchTags:
          - Key: environment
            Value: ^ci$
        metrics:
          - name: CPUUtilization
            statistics:
            - 'Average'
            period: 60
            length: 600
          - name: DiskReadOps
            statistics:
            - 'Average'
            period: 60
            length: 600
          - name: DiskWriteOps
            statistics:
            - 'Average'
            period: 60
            length: 600
          - name: StatusCheckFailed
            statistics:
            - 'Maximum'
            period: 60
            length: 600
      - region: us-east-2
        type: "elb"
        searchTags:
          - Key: KubernetesCluster
            Value: devtest.k8s.local
        metrics:
          - name: BackendConnectionErrors
            statistics:
            - Sum
            period: 60
            length: 600
          - name: HealthyHostCount
            statistics:
            - Average
            period: 60
            length: 600
          - name: HTTPCode_Backend_5XX
            statistics:
            - Sum
            period: 60
            length: 600
          - name: HTTPCode_ELB_5XX
            statistics:
            - Sum
            period: 60
            length: 600
          - name: Latency
            statistics:
            - Average
            - Maximum
            period: 60
            length: 600
          - name: RequestCount
            statistics:
            - Sum
            period: 60
            length: 600
          - name: SpilloverCount
            statistics:
            - Sum
            period: 60
            length: 600
          - name: SurgeQueueLength
            statistics:
            - Maximum
            - Sum
            period: 60
            length: 600
      - region: us-east-2
        type: "es"
        searchTags:
          - Key: Name
            Value: ci-logs
        metrics:
          - name: CPUUtilization
            statistics:
            - Average
            period: 60
            length: 600
          - name: FreeStorageSpace
            statistics:
            - Minimum
            period: 60
            length: 600
          - name: ClusterIndexWritesBlocked
            statistics:
            - Maximum
            period: 60
            length: 600
          - name: JVMMemoryPressure
            statistics:
            - Maximum
            period: 60
            length: 600
      - region: us-east-2
        type: "s3"
        searchTags:
          - Key: Environment
            Value: ci
        metrics:
          - name: BucketSizeBytes
            statistics:
            - Average
            period: 86400
            length: 172800
            disableTimestamp: true
          - name: NumberOfObjects
            statistics:
            - Average
            period: 86400
            length: 172800
            disableTimestamp: true
```

I am getting this in the metrics (as well as all the other ones requested):
```
# HELP aws_s3_info Help is not implemented yet.
# TYPE aws_s3_info gauge
aws_s3_info{name="arn:aws:s3:::<redacted 1>",tag_Environment="ci"} 0
aws_s3_info{name="arn:aws:s3:::<redacted 2>",tag_Environment="ci"} 0
# HELP aws_s3_number_of_objects_average Help is not implemented yet.
# TYPE aws_s3_number_of_objects_average gauge
aws_s3_number_of_objects_average{name="arn:aws:s3:::<redacted 1>"} 10468
aws_s3_number_of_objects_average{name="arn:aws:s3:::<redacted 2>"} 411004
```

The logs look clean:
```
2019/02/26 18:44:25 Parse config..
2019/02/26 18:44:25 Startup completed
```

**Comment by abuchananTW** [comment]
Running with debug, it appears to be fetching the data...
```
2019/02/28 17:24:34 {
  Dimensions: [{
      Name: "BucketName",
      Value: "<redacted>"
    },{
      Name: "StorageType",
      Value: "AllStorageTypes"
    }],
  EndTime: 2019-02-28 17:24:34.845663001 +0000 UTC m=+146.772204440,
  MetricName: "BucketSizeBytes",
  Namespace: "AWS/S3",
  Period: 86400,
  StartTime: 2019-02-26 17:24:34.845664572 +0000 UTC m=-172653.227794342,
  Statistics: ["Average"]
}
```

I should also note, this has been running for a few days without getting data

**Comment by abuchananTW** [comment]
Wondering if this is due to not sending a dimension on the CloudWatch call... Not sure if one is required for this one or not...

Is there a reason dimensions can't be added to discovered metrics?

**Comment by abuchananTW** [comment]
That does seem to be the issue. It requires sending a StorageType dimension (confirmed by command line queries).

I guess that brings me back to whether there is a reason dimensions can't be added to discovered metrics...

**Comment by tsupertramp** [comment]
Thanks for your awesome research that helps a lot to understand the underlying problem.

**Comment by tsupertramp** [comment]
Released in 0.13.0-alpha - Thanks for your work!
